libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1602,6 +1605,27 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         navigationSession.unregisterNavigationSessionStateObserver(navigationSessionStateObserver)
     }
 
+    /**
+     * Set a [RouteRefreshStatesObserver] to be notified of the various Route refresh states.
+     * Every new instance of [routeRefreshStatesObserver] substitute previous one.
+     *
+     * @param routeRefreshStatesObserver NavigationSessionStateObserver
+     */
+    @ExperimentalPreviewMapboxNavigationAPI
+    fun setRouteRefreshStateObserver(
```

Comment:
Can there be only one observer? In other similar cases we use a collection of them, don't we?

Reply:
Since this is a classically observable element as anything else in the SDK, let's use a collection, regardless of whether it's experimental or not.

Reply:
1 vs 2 🤔 😄 okay